File: frontend/bmp-commerce-web/src/types/venda.ts
export interface ItemVenda {
  produtoId: string
  produtoNome: string
  produtoSku: string
  quantidade: number
  precoVendaMomento: number
  subtotal: number
}

export interface Venda {
  id: string
  clienteId: string | null
  clienteNome: string | null
  usuarioNome: string
  dataHora: string
  total: number
  quantidadeItens: number
  cancelada: boolean
  itens: ItemVenda[]
}

export interface RegistrarVendaPayload {
  clienteId: string | null
  itens: { produtoId: string; quantidade: number }[]
}

File: backend/app/schemas/venda.py
"""Equivalente a Application/Operations/Vendas/VendaDtos.cs."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from app.schemas.common import CamelModel


class ItemVendaRequest(CamelModel):
    produto_id: UUID
    quantidade: int


class RegistrarVendaRequest(CamelModel):
    cliente_id: UUID | None = None
    itens: list[ItemVendaRequest]


class ItemVendaDto(CamelModel):
    produto_id: UUID
    produto_nome: str
    produto_sku: str
    quantidade: int
    preco_venda_momento: float
    subtotal: float


class VendaDto(CamelModel):
    id: UUID
    cliente_id: UUID | None
    cliente_nome: str | None
    usuario_nome: str
    data_hora: datetime
    total: float
    quantidade_itens: int
    cancelada: bool
    itens: list[ItemVendaDto]
